minicluster/src/main/java/org/apache/accumulo/miniclusterImpl/MiniAccumuloClusterControl.java

```diff
@@ -111,6 +127,53 @@ public void adminStopAll() throws IOException {
     }
   }
 
+  private static TExternalCompactionList getRunningCompactions(ClientContext context)
+      throws TException {
+    Optional<HostAndPort> coordinatorHost =
+        ExternalCompactionUtil.findCompactionCoordinator(context);
+    if (coordinatorHost.isEmpty()) {
+      throw new TTransportException("Unable to get CompactionCoordinator address from ZooKeeper");
+    }
+    CompactionCoordinatorService.Client client = ThriftUtil.getClient(
+        new CompactionCoordinatorService.Client.Factory(), coordinatorHost.get(), context);
+    try {
+      TExternalCompactionList running =
+          client.getRunningCompactions(TraceUtil.traceInfo(), context.rpcCreds());
+      return running;
+    } finally {
+      ThriftUtil.returnClient(client, context);
+    }
+  }
+
+  public synchronized void startCoordinator(Class<? extends CompactionCoordinator> coordinator)
+      throws IOException {
+    if (coordinatorProcess == null) {
+      coordinatorProcess = cluster
+          ._exec(coordinator, ServerType.COMPACTION_COORDINATOR, new HashMap<>()).getProcess();
+      UtilWaitThread.sleep(1000);
+      // Wait for coordinator to start
+      TExternalCompactionList metrics = null;
+      while (null == metrics) {
+        try {
+          metrics = getRunningCompactions(cluster.getServerContext());
+        } catch (Exception e) {
+          UtilWaitThread.sleep(250);
```

Comment:
Do all exceptions warrant merely sleeping and retrying? It seems to me that some exceptions might result in this never being able to succeed, and us leaving this thread waiting and retrying indefinitely.

Also, it'd be good to log the exception, at least at debug.

Reply:
I modified this to only catch TException in [851c2c3].